src/bal-converter/index.ts

```diff
@@ -0,0 +1,178 @@
+import type {
+  BanID,
+  BanCommonTopoID,
+  LangISO639v3,
+} from "../types/ban-generic-types.js";
+
+import type { Bal, BalAdresse, VoieNomForIsoCode } from "../types/bal-types.js";
+
+import type {
+  Ban,
+  BanAddress,
+  BanAddresses,
+  BanCommonToponyms,
+  BanCommonToponym,
+} from "../types/ban-types.js";
+
+import {
+  getAddressIdsReport,
+  createAddresses,
+  updateAddresses,
+  deleteAddresses,
+  getCommonToponymIdsReport,
+  createCommonToponyms,
+  updateCommonToponyms,
+  deleteCommonToponyms,
+} from "../ban-api/index.js";
+
+import { csvToJson } from "./utils.js";
+
+const getIds = (balAdresse: BalAdresse) => {
+  const [banID, banCommonTopoID] = balAdresse.uid_adresse.split("/");
+  return {
+    banID,
+    banCommonTopoID,
+  };
+};
+
+const balAddrToBanAddr = (
+  balAdresse: BalAdresse,
+  oldBanAddress?: BanAddress
+): BanAddress => {
+  const { banID, banCommonTopoID } = getIds(balAdresse);
+  return {
+    ...(oldBanAddress || {}),
+    id: banID,
+    codeCommune: balAdresse.commune_insee,
+    idVoie: banCommonTopoID,
+    numero: balAdresse.numero,
+    positions: [
+      // Old positions
+      ...(oldBanAddress?.positions || []),
+      {
+        type: balAdresse.position,
+        geometry: {
+          type: "Point",
+          coordinates: [balAdresse.long, balAdresse.lat],
+        },
+      },
+    ],
+    dateMAJ: balAdresse.date_der_maj,
+  };
+};
+
+const balTopoToBanTopo = (
+  balAdresse: BalAdresse,
+  oldBanCommonToponym?: BanCommonToponym
+): BanCommonToponym => {
+  const { banCommonTopoID } = getIds(balAdresse);
+  const isoCodeFromBalNomVoie = (key: LangISO639v3) => key.trim().split("_")[2];
+  const labels = {
+    fr: balAdresse.voie_nom,
+    ...Object.fromEntries(
+      (
+        Object.keys(balAdresse).filter((key) =>
+          key.startsWith("voie_nom_")
+        ) as VoieNomForIsoCode[]
+      ).map((key) => [isoCodeFromBalNomVoie(key), balAdresse[key]])
+    ),
+  };
+
+  return {
+    ...(oldBanCommonToponym || {}),
+    id: banCommonTopoID,
+    codeCommune: balAdresse.commune_insee,
+    label: Object.entries(labels).map(([isoCode, value]) => ({
+      isoCode,
+      value,
+    })), // TODO: rename key to 'labels'
+    type: { value: "voie" }, // TODO: How to get the type from the BAL?
+    geometry: {
+      type: "Point",
+      coordinates: [balAdresse.long, balAdresse.lat],
+    },
+    dateMAJ: balAdresse.date_der_maj,
+  };
+};
+
+const balToBan = (bal: Bal): Ban => {
+  const ban = bal.reduce(
+    (acc: Ban, balAdresse: BalAdresse) => {
+      const { banID, banCommonTopoID } = getIds(balAdresse);
+      const banIdContent = balAddrToBanAddr(balAdresse, acc.addresses?.[banID]);
+      const banCommonTopoIdContent = balTopoToBanTopo(
+        balAdresse,
+        acc.commonToponyms?.[banCommonTopoID]
+      );
+      const districtID = banIdContent.codeCommune;
```

Comment:
```suggestion
      const districtID = `${banIdContent.codeCommune}`;
```
Le district ID est de type string dans le format BAN